python/shotgun_globals/date_time.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import datetime
+
+def create_human_readable_timestamp(dt, postfix=""):
+    """
+    Return the time represented by the argument as a string where the date portion is
+    displayed as "Yesterday", "Today", or "Tomorrow" if appropriate.
+
+    By default just the date is displayed, but additional formatting can be appended
+    by using the postfix argument.
+
```

Comment:
Heads up that the activity stream has got a similar logic:

https://github.com/shotgunsoftware/tk-framework-qtwidgets/blob/aa8ed8c49814dfea96c0cf5863d598de97a42a39/python/activity_stream/widget_activity_stream_base.py#L106

This is trying to make the string representation as simple as possible:

- if the time is more than one year ago, display on the form `26 June 2012`
- if the time is more than one week ago, display on the form `26 June`
- if the time is more than one day ago, display it as day of week - e.g. `Sunday`
- if the time is from earlier today, display it as a timestamp, e.g. `23:22`

Maybe the use case is different for this type of method? Maybe at some point we should add this logic to the shotgun global too?

btw, i noticed that that code has got the same potential UTC unixtime stamp issue, so maybe our unix times are actually not stored in UTC as they ideally should be, but stored in the local tz.... Maybe something to discuss.